src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -1163,76 +1163,78 @@ void ShenandoahFreeSet::prepare_to_rebuild(size_t &young_cset_regions, size_t &o
 
 void ShenandoahFreeSet::rebuild(size_t young_cset_regions, size_t old_cset_regions, bool have_evacuation_reserves) {
   shenandoah_assert_heaplocked();
-  size_t young_reserve, old_reserve;
-  size_t region_size_bytes = ShenandoahHeapRegion::region_size_bytes();
+  size_t young_reserve(0), old_reserve(0);
+
+  if (!_heap->mode()->is_generational()) {
+    young_reserve = (_heap->max_capacity() / 100) * ShenandoahEvacReserve;
+    old_reserve = 0;
+  } else {
+    compute_young_and_old_reserves(young_cset_regions, old_cset_regions, have_evacuation_reserves,
+                                   &young_reserve, &old_reserve);
+
+  }
+
+  reserve_regions(young_reserve, old_reserve);
+  _free_sets.establish_alloc_bias(OldCollector);
+  _free_sets.assert_bounds();
+  log_status();
+}
+
+void ShenandoahFreeSet::compute_young_and_old_reserves(size_t young_cset_regions, size_t old_cset_regions,
+                                                       bool have_evacuation_reserves, size_t* young_reserve_result,
+                                                       size_t* old_reserve_result) const {
```

Comment:
Might read more "naturally" if args declared as:
```
size_t& young_reserve_result, size_t& old_reserve_result

Reply:
Hmm, I usually see 'result' parameters like this passed as pointers in the JDK. It makes it clearer at the call site. Or did you mean to format them on the same line?